tests/unit/metrics/test_metrics_automated.py

```diff
@@ -0,0 +1,334 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Automated testing framework for LightEval metrics.
+
+This module provides a simple way to test metrics by providing input/output pairs.
+You can define test cases with expected inputs and outputs, and the framework will
+automatically run them and verify the results.
+"""
+
+import copy
+import json
+import logging
+import math
+from dataclasses import field
+from pathlib import Path
+from typing import Any
+
+from pydantic import BaseModel
+
+from lighteval.metrics.metrics import Metrics
+from lighteval.models.model_output import ModelResponse
+from lighteval.tasks.requests import Doc
+
+
+logger = logging.getLogger(__name__)
+
+
+class MetricTestCase(BaseModel):
+    """A test case for a metric with input and expected output."""
+
+    name: str
+    metric_class: str
+    metric_params: dict[str, Any] = field(default_factory=dict)
+    doc: dict[str, Any]
+    model_response: dict[str, Any]
+    expected_output: dict[str, float]
+    tolerance: float = 1e-2
+    description: str | None = None
+
+
+class CorpusLevelMetricTestCase(BaseModel):
+    """A test case for a corpus level metric with input and expected output."""
+
+    name: str
+    metric_class: str
+    metric_name: str
+    metric_params: dict[str, Any] = field(default_factory=dict)
+    docs: list[dict[str, Any]]
+    model_responses: list[dict[str, Any]]
+    expected_output: float
+    tolerance: float = 1e-2
+    description: str | None = None
+
+
+class MetricTestSuite(BaseModel):
+    """A collection of test cases for metrics."""
+
+    name: str
+    test_cases: list[MetricTestCase | CorpusLevelMetricTestCase]
+    corpus_level: bool = False
+    description: str | None = None
+
+
+class AutomatedMetricTester:
+    """Automated testing framework for LightEval metrics."""
+
+    # Mapping of metric names to Metrics enum values
+    METRIC_CLASSES = {
```

Comment:
Could you do this automatically ? 
METRIC_CLASSES = {metric.value.name for metric in Metrics} 

(or something like this)

Would allow to make sure all metrics have a corresponding test

Reply:
I would add a "skip list" then, like `for metric in Metrics if metric not in SKIPPED_METRICS`

Reply:
That way we're sure every new metric has tests added